firmware/src/main_debug_noharp.cpp

```diff
@@ -0,0 +1,79 @@
+#include <pico/stdlib.h> // for uart printing
+#include <cstring>
+#include <stdio.h>
+#include <config.h>
+#include <pico/stdio_usb.h>
+#include <stdio.h>
+#include "Adafruit_BME680_rpi.h"
+
+
+void set_led_state(bool enabled)
+{
+    if (enabled)
+    {
+        gpio_init(LED_PIN);
+        gpio_set_dir(LED_PIN, true); // true for output
+        gpio_put(LED_PIN, 0);
+    }
+    else
+        gpio_deinit(LED_PIN);
+}
+
+Adafruit_BME680 bme688(spi0,BME688_PICO_PIN,BME688_POCI_PIN,BME688_SCK_PIN,BME688_CS_PIN,true);
+
+// copied from https://github.com/adafruit/Adafruit_BME680/blob/master/examples/bme680test/bme680test.ino
+void loop(Adafruit_BME680& bme) {
+  uint32_t ping = time_us_32();
+  if (! bme.performReading()) {
+    printf("Failed to perform reading :(\r\n");
+    return;
+  }
+  uint32_t pong = time_us_32();
```

Comment:
Slightly pedantic, but let's change this to `curr_time_us` and `start_time_us` or similar since `ping` and `pong` are affiliated with [double buffering](https://embedded.fm/blog/2017/3/21/ping-pong-buffers).

Overall, adding this file looks like a good option for compiling and testing device functionality minus all the harp stuff without a cluster of `#ifdef` macros sprinkled everywhere.